docs/c-ffi/calling-c.md

```diff
@@ -97,20 +89,29 @@ let size = Winsize
 env.out.print(size.height.string())
 ```
 
-## FFI functions raising errors
-
-FFI functions can raise Pony errors. Functions in existing C libraries are very unlikely to do this, but support libraries specifically written for use with Pony may well do.
+### Variadic C functions
 
-FFI calls to functions that __might__ raise an error __must__ mark it as such by adding a ? after the arguments. For example:
+Some C functions are variadic, that is, they can take a variable number of parameters. To interact with these functions, you should also specify that fact in the FFI signature:
 
 ```pony
-@os_send[U64](_event, data.cstring(), data.size()) ? // May raise an error
+use @printf[I32](fmt: Pointer[U8] tag, ...)
+// ...
+let run_ns: I64 = _current_t - _last_t
+let rate: I64 = (_partial_count.i64() * 1_000_000_000) / run_ns
+@printf("Elapsed: %lld,%lld\n".cstring(), run_ns, rate)
 ```
 
-If a signature declaration is used then that must be marked as possibly raising an error in the same way. The FFI call site must mark it as well.
+In the example above, the compiler will type-check the first argument to `printf`, but will not be able to check any other argument, since it lacks the necessary type information. It is __very__ important that you use `...` in the FFI signature if the corresponding C function is variadic: if you don't, the compiler might generate a program that is incorrect or crash, depending on the target platform.
```

Comment:
Small suggestion to make it a bit more clear how creepy and subtle this problem can be:

```suggestion
In the example above, the compiler will type-check the first argument to `printf`, but will not be able to check any other argument, since it lacks the necessary type information. It is __very__ important that you use `...` in the FFI signature if the corresponding C function is variadic: if you don't, the compiler might generate a program that is incorrect or crash on some platforms while appearing to work correctly on others.
```